conformance/tests/httproute-matching-combinations.go

```diff
@@ -0,0 +1,144 @@
+/*
+Copyright 2023 The Kubernetes Authors.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package tests
+
+import (
+	"testing"
+
+	"k8s.io/apimachinery/pkg/types"
+
+	"sigs.k8s.io/gateway-api/conformance/utils/http"
+	"sigs.k8s.io/gateway-api/conformance/utils/kubernetes"
+	"sigs.k8s.io/gateway-api/conformance/utils/suite"
+)
+
+func init() {
+	ConformanceTests = append(ConformanceTests, HTTPRouteMatchingCombinations)
+}
+
+var HTTPRouteMatchingCombinations = suite.ConformanceTest{
+	ShortName:   "HTTPRouteMatchingCombinations",
+	Description: "An HTTPRoute containing combinations of multiple match types",
+	Manifests:   []string{"tests/httproute-matching-combinations.yaml"},
+	Features: []suite.SupportedFeature{
+		suite.SupportGateway,
+		suite.SupportHTTPRoute,
+		suite.SupportHTTPRouteQueryParamMatching,
+		suite.SupportHTTPRouteMethodMatching,
+	},
```

Comment:
Similar to how the Redirect tests were split up into separate tests for each extended feature combined with all core features, it could be helpful to have subsets of this available. For the record, I don't think that needs to block this PR from merging, but could be a good follow up.

Reply:
I went ahead and split it up here itself, since we're anyway waiting for LGTM. I wasn't aware we try to maintain such a pattern. But yes it does make sense. 

@robscott -- I guess this implies that if one wishes to add tests that cover ALL match types, that would go in it's own file/test. Will see if that is needed in a separate PR.

Reply:
Yep, I think separate files are _probably_ sufficient here, but don't feel too strongly either way.